pinot-core/src/main/java/org/apache/pinot/core/common/PinotRuntimeException.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.common;
+
```

Comment:
I'd be good to discuss how to integrate the approach with other exceptions and re-organize exceptions thrown by the engine. There are many more places that don't report conversion errors properly (e.g. plenty of parse calls for function arguments that won't report which argument they failed for).

Reply:
I think this is a really nice improvement! IMO we could add similar custom runtime exceptions for different classes of processing errors as a subsequent enhancement. Although in that case it might be nice to name this class something more specific than `PinotRuntimeException`, WDYT?

Reply:
I completely agree we need to have our own family of exceptions, but I think it will need more discussion.

My initial suggestion is to reserve `PinotRuntimeException` name for a marker exception that extends `RuntimeException`, then add another for queries (something like `QueryException`) and then 


```mermaid
---
config:
  look: handDrawn
  theme: neutral
---
classDiagram
    RuntimeException
    note for RuntimeException "From JDK"

    PinotRuntimeException
    note for PinotRuntimeException "All new Pinot exceptions"

    QException
    note for QException "All new Pinot exceptions"

    QException <|-- ParsingQException
    QException <|-- OptimizationQException

    ExecutionQException
    note for ExecutionQException "Exceptions thrown by Operators"

    RuntimeException <|-- PinotRuntimeException
    PinotRuntimeException <|-- QException
    QException <|-- ExecutionQException
    ExecutionQException <|-- ColumnExecutionQException
    
    class ExecutionQException {
        +stage
    }

    class ColumnExecutionQException {
        +table
        +column
    }
    note for ColumnExecutionQException "Execution exception dealing with a column"
```